recipes/strawberryperl/all/conanfile.py

```diff
@@ -1,39 +1,54 @@
+    license = ("Artistic-1.0", "GPL-1.0")
     homepage = "http://strawberryperl.com"
     url = "https://github.com/conan-io/conan-center-index"
-    topics = ("conan", "installer", "perl", "windows")
-    settings = "os", "arch"
-    short_paths = True
+    topics = ("installer", "perl", "windows")
+    settings = "os", "arch", "compiler", "build_type"
+
+    def layout(self):
+        self.folders.build = "build"
+
+    def package_id(self):
+        del self.info.settings.compiler
+        del self.info.settings.build_type
 
-    def configure(self):
-        if self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("Only windows supported for Strawberry Perl.")
+    def validate(self):
+        if self.info.settings.os != "Windows":
+            raise ConanInvalidConfiguration("Strawberry Perl is only intended to be used on Windows.")
 
     def build(self):
-        arch = str(self.settings.arch)
-        tools.get(**self.conan_data["sources"][self.version][arch])
+        get(self, **self.conan_data["sources"][self.version][str(self.settings.arch)], destination=self.build_folder)
 
     def package(self):
-        self.copy(pattern="License.rtf*", dst="licenses", src="licenses")
-        self.copy(pattern="*", src=os.path.join("perl", "bin"), dst="bin")
-        self.copy(pattern="*", src=os.path.join("perl", "lib"), dst="lib")
-        self.copy(pattern="*", src=os.path.join("perl", "vendor", "lib"), dst="lib")
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        copy(self, pattern="License.rtf*", src=os.path.join(self.build_folder, "licenses"), dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, pattern="*", src=os.path.join(self.build_folder, "perl", "bin"), dst=os.path.join(self.package_folder, "bin"))
+        copy(self, pattern="*", src=os.path.join(self.build_folder, "perl", "lib"), dst=os.path.join(self.package_folder, "lib"))
+        copy(self, pattern="*", src=os.path.join(self.build_folder, "perl", "vendor", "lib"), dst=os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
         self.cpp_info.libdirs = []
         self.cpp_info.includedirs = []
 
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: %s" % bin_path)
-        self.env_info.PATH.append(bin_path)
+        # TODO remove once conan v2 is the only support and recipes have been migrated
+        if Version(conan_version) < "2.0.0-beta":
+            bin_path = os.path.join(self.package_folder, "bin")
+            self.env_info.PATH.append(bin_path)
 
-        self.user_info.perl = os.path.join(self.package_folder, "bin", "perl.exe").replace("\\", "/")
+        perl_path = os.path.join(self.package_folder, "bin", "perl.exe").replace("\\", "/")
+        self.conf_info.define("user.strawberryperl:perl", perl_path)
```

Comment:
Doesn't exist in conf list, is it allowed or even intended in conan philosophy?

Reply:
#13490 :)

Reply:
thanks